patches/master_patch.patch

```diff
@@ -1831,6 +1910,179 @@ index a433afd0217830b49ddb3381ad55afeae2381840..ddcca749666f956c6317676dbcdd48da
+index 551953165f93c8d8e4772f0e905f1c4819aaf3b5..caf6df58e858580b165e6ff689572a3d26525510 100644
+--- a/net/log/net_log_event_type_list.h
++++ b/net/log/net_log_event_type_list.h
+@@ -2108,6 +2108,12 @@ EVENT_TYPE(SOCKS5_GREET_WRITE)
+ // The time spent waiting for the "greeting" response from the SOCKS server.
+ EVENT_TYPE(SOCKS5_GREET_READ)
+ 
++// The time spent sending authentication to the SOCKS server
++EVENT_TYPE(SOCKS5_AUTH_WRITE)
++
++// The time spent waiting for the authentication response from the SOCKS server
++EVENT_TYPE(SOCKS5_AUTH_READ)
++
+ // The time spent sending the CONNECT request to the SOCKS server.
+ EVENT_TYPE(SOCKS5_HANDSHAKE_WRITE)
+ 
+diff --git a/net/proxy/proxy_server.cc b/net/proxy/proxy_server.cc
+index ed81ebbb26b4f439d618c115b32f49249d9284a6..ebfa79de63e40539a09d072a62c8522fef7cf0f4 100644
+--- a/net/proxy/proxy_server.cc
++++ b/net/proxy/proxy_server.cc
+@@ -8,6 +8,7 @@
+ 
+ #include "base/strings/string_util.h"
+ #include "base/trace_event/memory_usage_estimator.h"
++#include "net/base/url_auth_util.h"
+ #include "net/base/url_util.h"
+ #include "net/http/http_util.h"
+ 
+@@ -232,7 +233,7 @@ ProxyServer ProxyServer::FromSchemeHostAndPort(
+     std::string host;
+     int port = -1;
+     // If the scheme has a host/port, parse it.
+-    bool ok = ParseHostAndPort(begin, end, &host, &port);
++    bool ok = ParseAuthHostAndPort(begin, end, &host, &port);
+     if (!ok)
+       return ProxyServer();  // Invalid -- failed parsing <host>[":"<port>]
+ 
+diff --git a/net/socket/socks5_client_socket.cc b/net/socket/socks5_client_socket.cc
+index 19cdfddbe0c533b18f57bfcdd25ff9958ac724dc..c99f6642e0204fc5c2c6e745dcce67473e2b8188 100644
+--- a/net/socket/socks5_client_socket.cc
++++ b/net/socket/socks5_client_socket.cc
+@@ -245,6 +245,9 @@ int SOCKS5ClientSocket::DoLoop(int last_io_result) {
+         net_log_.EndEventWithNetErrorCode(NetLogEventType::SOCKS5_GREET_READ,
+                                           rv);
+         break;
++      case STATE_AUTH:
++        rv = DoAuth(rv);
++        break;
+       case STATE_HANDSHAKE_WRITE:
+         DCHECK_EQ(OK, rv);
+         net_log_.BeginEvent(NetLogEventType::SOCKS5_HANDSHAKE_WRITE);
+@@ -274,8 +277,6 @@ int SOCKS5ClientSocket::DoLoop(int last_io_result) {
+   return rv;
+ }
+ 
+-const char kSOCKS5GreetWriteData[] = { 0x05, 0x01, 0x00 };  // no authentication
+-
+ int SOCKS5ClientSocket::DoGreetWrite() {
+   // Since we only have 1 byte to send the hostname length in, if the
+   // URL has a hostname longer than 255 characters we can't send it.
+@@ -285,8 +286,12 @@ int SOCKS5ClientSocket::DoGreetWrite() {
+   }
+ 
+   if (buffer_.empty()) {
+-    buffer_ = std::string(kSOCKS5GreetWriteData,
+-                          arraysize(kSOCKS5GreetWriteData));
++    const char greeting[] = {
++      0x05,			// SOCKS version
++      0x01,			// number of authentication methods
++      auth_method(),
++    };
++    buffer_ = std::string(greeting, sizeof(greeting));
```

Comment:
does this still work for non-auth socks proxies?

Reply:
Should.  In SOCKS5ClientSocket, auth_method() returns zero, so `greeting` is identical to the old `kSOCKS5GreetWriteData`.  The tests that I discarded from the patch to minimize it used to run with this new logic.  Haven't tried with C65 but I don't see why it should be different.